src/data-access/repositories/mongo/mongo-entity-converter.ts

```diff
@@ -209,15 +280,25 @@ export class MongoEntityConverter {
   }
 
   public convertToMongoPart(part: Part): MongoPart {
+    let timings: PartTimings | undefined = undefined
+    try {
+      timings = part.getTimings()
+    } catch (error) {
+      if ((error as Exception).errorCode !== ErrorCode.UNSUPPORTED_OPERATION) {
+        throw error
+      }
+    }
     return {
       _id: part.id,
       isPlanned: part.isPlanned,
       expectedDuration: part.expectedDuration,
       title: part.name,
       segmentId: part.getSegmentId(),
-      _rank: part.rank,
+      _rank: part.getRank(),
       isOnAir: part.isOnAir(),
       isNext: part.isNext(),
+      isUnsynced: part.isUnsynced(),
+      timings,
       endState: part.getEndState()
     } as MongoPart
```

Comment:
It would be nice if we could get rid of these type assertion.

Reply:
I agree, but that is not the scope of this PR.